src/pointer-analysis/dereference.h

```diff
@@ -199,13 +199,53 @@ class dereferencet
   virtual ~dereferencet() = default;
 
   /** The different ways in which a pointer may be accessed. */
-  typedef enum
+  struct modet
```

Comment:
Can you make this a class with a proper constructor?

Reply:
> designated initializers

Done. That's much better. I don't like pessimizing the classification of a type; it avoids optimization opportunities and (to me) is less clear.

Reply:
Windows:
> error C7555: use of designated initializers requires at least '/std:c++20'